keps/0014-pull-request-process.md

```diff
@@ -0,0 +1,129 @@
+---
+kep-number: 0014
+title: Pull Request Process
+authors:
+  - "@jbarrick-mesosphere"
+owners:
+  - "@jbarrick-mesosphere"
+  - "@orsenthil"
+editor: TBD
+creation-date: 2019-07-16
+last-updated: 2019-07-16
+status: provisional
+see-also:
+  - KEP-0004
+  - KEP-0008
+---
+
+# Pull Request Process
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Goals](#goals)
+* [Proposal](#proposal)
+   * [Issues and pull requests](#issues-and-pull-requests)
+   * [Pull requests](#pull-requests)
+   * [Test environment](#test-environment)
+* [User Stories](#user-stories)
+   * [Issue reporter](#issue-reporter)
+   * [Outside contributor](#outside-contributor)
+   * [Trusted contributor](#trusted-contributor)
+   * [Repository administrator](#repository-administrator)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+In order to ensure that contributors to the KUDO project have a good experience, we have a streamlined pull request review and merging process using Prow.
+
+## Goals
+
+* Automate and enforce labeling of pull requests and issues (e.g., size).
+* Automate pull request and issue life cycles.
+* Automatically request pull request review from the correct people.
+* Allow contributors to restrict who is allowed to run tests.
+* Not be limited by public CI services (queue times, concurrency, resources).
+* Use CNCF-standard tooling for our processes.
+
+## Proposal
+
+A Prow cluster will be deployed in Kubernetes for running CI and PR/issue automation. This cluster lives at https://prow.toolsinfra.mesosphe.re/ and will take action on Github as the `ci-mergebot` user. KEP-0004 outlines the infrastructure required.
```

Comment:
Could this be an isolated KUDO Community resource of Prow and not attached to Mesosphe.re? As discussed here I think it would make more sense to make it isolated. https://www.youtube.com/watch?v=eW0qfhEVTTY&feature=youtu.be&t=1999

Reply:
So here's what I'm thinking:

* Let's run it at https://prow.kudo.dev/.
* It will be for the `kudobuilder/` organization only.
* Run it on GKE.
* The cluster configuration is open source and lives in `kudobuilder/test-infra`.